File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int target;

bool cmp(int a, int b){
    if(abs(a - target) != abs(b - target))
        return abs(a - target) < abs(b - target);
    
    return a > b;
}

vector<int> solution(vector<int> numlist, int n) {
    target = n;
    
    sort(numlist.begin(), numlist.end(), cmp);
    
    return numlist;
}

File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.js
function solution(numlist, n) {
    // 비교 함수 정의
    const cmp = (a, b) => {
        const diffA = Math.abs(a - n);
        const diffB = Math.abs(b - n);
        
        if (diffA !== diffB) {
            return diffA - diffB; // 오름차순 정렬
        }
        
        return b - a; // 내림차순 정렬
    };
    
    // 배열 정렬
    numlist.sort(cmp);
    
    return numlist;
}